### PR TITLE
Refactored combiner registry and broke circular dep with schema

### PR DESCRIPTION
**Pull request by tgaddair**

### CODE REVIEW

**Review by ksbrar — COMMENTED** [review]
So can we get rid of this now? https://github.com/ludwig-ai/ludwig/blob/397e46b44a2e39fc8f6c81b94b398c491ff671c0/ludwig/config_validation/validation.py#L26

**Comment by tgaddair** [comment]
@ksbrar good call!

**Review by connor-mccorm — APPROVED** [review]
LGTM!

File: tests/integration_tests/test_custom_components.py [hunk 1]
```diff
@@ -61,7 +55,15 @@ class CustomLossConfig(BaseLossConfig):
     type: str = "custom_loss"
 
 
-@register_combiner(name="custom_test")
+@register_combiner_schema("custom_combiner")
```
Comment (ksbrar) [inline tests/integration_tests/test_custom_components.py:58]: Is this going to end up in the combiner registry? (No, right? - since it's under `tests`)

Comment (tgaddair) [reply tests/integration_tests/test_custom_components.py:58]: That's right.

**Review by ksbrar — APPROVED** [review]
LGTM. Looks like the tests are OOMing 😞

**Comment by tgaddair** [comment]
Yep, gonna see what's up with that :(

**Pull request merged**